Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -369,6 +377,30 @@ inline bool ShouldDraw() {
         return CurrentGState().ShouldDraw();
     }
 
+    inline void PushBeginDraw() {
+        if ((_beginEndDrawDepth)++ == 0) {
+            deviceContext->BeginDraw();
+        }
+    }
+
+    inline void PopEndDraw() {
+        if (--(_beginEndDrawDepth) == 0) {
+            THROW_IF_FAILED(deviceContext->EndDraw());
+        }
+    }
+
+    inline void EscapeBeginEndDrawStack() {
+        if ((_beginEndDrawDepth > 0) && ((_escapeBeginEndDrawDepth)++ == 0)) {
+            THROW_IF_FAILED(deviceContext->EndDraw());
```

Comment:
The recommendation is wrong, we agreed not to FAIL_FAST at arbitrary points in the CG code and instead throw.  Any exceptions must be caught at the "C" entry point, where we decide either to return an error code/null or fail-fast if it is void.  

---
In reply to: [97695158](https://github.com/Microsoft/WinObjC/pull/1750#discussion_r97695158) [](ancestors = 97695158,97468410)